opennms-webapp-rest/src/test/java/org/opennms/web/rest/v2/MonitoringLocationRestServiceIT.java

```diff
@@ -164,4 +172,74 @@ public void testEventOnCreationAndDeletion() throws Exception {
         // delete the one with polling packages
         sendRequest(DELETE, "/monitoringLocations/location2", 204);
     }
+
+    @Test
+    @Transactional
+    public void testLocationLimits() throws Exception {
+
+        final Integer DEFAULT_LIMIT = 10;
+        final Integer LOCATION_COUNT = 15;
+        final ObjectMapper MAPPER = new ObjectMapper();
+
+        this.eventIpcManager.getEventAnticipator().reset();
+
+        // There is one default location
+        LOG.info("Adding locations...");
+        for (int i = 0; i < LOCATION_COUNT - 1; i++) {
+            OnmsMonitoringLocation loc = new OnmsMonitoringLocation();
+            loc.setLocationName(String.format("Location-%05d",i));
+            loc.setMonitoringArea(String.format("LocationArea-%05d", i));
+            sendData(POST, MediaType.APPLICATION_XML,"/monitoringLocations", JaxbUtils.marshal(loc), 201);
+            this.eventIpcManager.getEventAnticipator().verifyAnticipated();
+        }
+
+        // null limit (empty parameters) should return default
+        assertThat(MAPPER.readValue(sendRequest(GET, "/monitoringLocations",
```

Comment:
Can you replace `sendRequest` with `getJsonObject`?

Reply:
The other test cases are using sendRequest so I thought this was the way to do it?

Reply:
`getJsonObject` is the same but already parses the data.

Reply:
It seems getJsonObject ignores the parameter map and just returns the default count of 10

Eg this fails with expected 1 but was 10

assertThat(getJsonObject(MAPPER,"/monitoringLocations",
                Map.of("limit","1"),200,OnmsMonitoringLocationDefinitionList.class).getCount(),is(1));

I have looked for other examples of getJsonObject being called with a populated map but can't find any. The only solution I have found is to put the limit in the url

Reply:
Thanks for the API fix Dustin, getJsonObject is working as expected now

Reply:
I have tried using the suggested changes to org.opennms.core.test.rest.AbstractSpringJerseyRestTestCase createRequest and it solves the problem for this specific test case but it has a huge knock on affect which I can't resolve, I have spent far more time on this than I think is necessary for a very simple test case which works perfectly using the standard sendRequest used by other tests. Which way would you like me to move forward?